mithril-core/src/stm_batch_compat.rs

```diff
@@ -0,0 +1,932 @@
+//! Batch proof helper.
+use crate::dense_mapping::ev_lt_phi;
+use crate::error::{AggregationError, StmSignatureError};
+use crate::key_reg::ClosedKeyReg;
+use crate::merkle_tree::{BatchPath, MTLeaf, MerkleTreeCommitmentBatchCompat};
+use crate::multi_sig::{Signature, SigningKey, VerificationKey};
+use crate::stm::{Index, Stake, StmClerk, StmInitializer, StmParameters, StmVerificationKey};
+use crate::RegisterError;
+use blake2::digest::{Digest, FixedOutput};
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::convert::{From, TryFrom, TryInto};
+use std::hash::{Hash, Hasher};
+use std::marker::PhantomData;
+
+/// Wrapper for StmClerk
+pub type StmClerkBatchCompact<D> = StmClerk<D>;
+
+/// Wrapper for StmInitializer
+pub type StmInitializerBatchCompat = StmInitializer;
+
+/// Participant in the protocol can sign messages.
+/// This instance can only be generated out of an `StmInitializer` and a `ClosedKeyReg`.
+/// This ensures that a `MerkleTree` root is not computed before all participants have registered.
+#[derive(Debug, Clone)]
+pub struct StmSignerBatchCompat<D: Digest> {
+    mt_index: u64,
+    stake: Stake,
+    params: StmParameters,
+    sk: SigningKey,
+    vk: StmVerificationKey,
+    closed_reg: ClosedKeyReg<D>,
+}
```

Comment:
This is the exact same structure as `StmSigner`. Why do we need a new structure?